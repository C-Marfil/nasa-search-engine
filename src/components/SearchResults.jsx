import React from "react";
import { useParams, Link } from "react-router-dom";
import PropTypes from "prop-types";
import "../styles/search-results.css";

const SearchResults = ({ results }) => {
  const { q } = useParams();

  if (!results.length) {
    return <p>Search images by using the text field above</p>;
  }
  return (
    <>
      <div className="images-container">
        {results.map((e) => {
          return (
            <Link to={`/${q}/image-detail`} state={{ src: `${e}` }}>
              <img key={e} className="image" src={e} alt="space-is-pretty" />
            </Link>
          );
        })}
      </div>
    </>
  );
};
SearchResults.defaultProps = {
  length: 0,
};

SearchResults.propTypes = {
  results: PropTypes.arrayOf(PropTypes.string).isRequired,
  length: PropTypes.number,
};

export default SearchResults;
