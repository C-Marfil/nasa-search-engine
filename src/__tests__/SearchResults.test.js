import { React, render, screen } from "@testing-library/react";
import SearchResults from "../components/SearchResults";

describe("Search Results component", () => {
  const validProps = {
    results: ["http//1", "http//2"],
    length: 2,
  };
  it("Snapshot test", () => {
    const { asFragment } = render(
      <SearchResults results={validProps.results} />
    );

    expect(asFragment()).toMatchSnapshot();
  });

  it("renders elements correctly", () => {
    render(<SearchResults results={validProps.results} />);

    expect(screen.getByText("Search Results")).toBeInTheDocument();
    expect(screen.getAllByRole("img")).toHaveLength(2);
  });
});
