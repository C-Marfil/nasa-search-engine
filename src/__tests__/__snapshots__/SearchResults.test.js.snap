// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Search Results component Snapshot test 1`] = `
<DocumentFragment>
  <p>
    Search Results
  </p>
  <img
    alt="space -is-pretty"
    src="http//1"
  />
  <img
    alt="space -is-pretty"
    src="http//2"
  />
</DocumentFragment>
`;
